ipatests/test_webui/ui_driver.py

```diff
@@ -362,6 +362,19 @@ def wait_for_request(self, implicit=0.2, n=1, d=0):
             self.wait()
         self.wait(d)
 
+    def wait_while_working(self, widget, implicit=0.2):
+        """
+        Wait while working widget active
+        """
+
+        working_widget = self.find('.working-widget', By.CSS_SELECTOR, widget)
```

Comment:
Could this `wait_while_working()` method be more generic? Something like receiving the CSS selector as an argument? Not sure if this is reusable or not. If it's too specific I believe we are good then.

Reply:
There is also different condition for exit the loop. So I wouldn't make over parameterized methods because it will be hard to debug.